Kernel/Devices/IDEDiskDevice.cpp

```diff
@@ -83,9 +83,25 @@ NonnullRefPtr<IDEDiskDevice> IDEDiskDevice::create()
     return adopt(*new IDEDiskDevice);
 }
 
+NonnullRefPtr<IDEDiskDevice> IDEDiskDevice::create(bool slave)
+{
+    return adopt(*new IDEDiskDevice(slave));
+}
+
 IDEDiskDevice::IDEDiskDevice()
     : IRQHandler(IRQ_FIXED_DISK)
     , m_io_base(0x1f0)
+    , m_second_drive(false)
```

Comment:
Initialize it in the header along with the other vars.

Also, why not merge the two overloads of the constructor together with a default value on the parameter?

Reply:
> Also, why not merge the two overloads of the constructor together with a default value on the parameter?

This was to maintain comparability with the current codebase, also assuming that instantiating a new `IDEDiskDevice` would automatically be the master on the channel. However, I do agree that it makes more sense to just have the one constructor. I'll modify it now.